--- a/usr/share/initramfs-tools/scripts/local-top/cryptroot	2018-06-15 01:36:11.216599204 +0200
+++ b/usr/share/initramfs-tools/scripts/local-top/cryptroot	2018-06-15 01:35:54.188599509 +0200
@@ -77,6 +77,8 @@
 	cryptveracrypt=""
 	cryptrootdev=""
 	cryptdiscard=""
+	crypttpm=""
+	crypttpmpcr=""
 	CRYPTTAB_OPTIONS=""
 
 	local IFS=" ,"
@@ -152,6 +154,13 @@
 		discard)
 			cryptdiscard="yes"
 			;;
+		tpm=*)
+			crypttpm="${x#tpm=}"
+			;;
+		tpmpcr=*)
+			crypttpmpcr="${x#tpmpcr=}"
+			crypttpmpcr="${crypttpmpcr/:/,}"
+			;;
 		esac
 		PARAM="${x%=*}"
 		if [ "$PARAM" = "$x" ]; then
@@ -300,6 +315,12 @@
 	fi
 	if /sbin/cryptsetup isLuks ${cryptheader:-$cryptsource} >/dev/null 2>&1; then
 		cryptopen="$cryptopen open --type luks $cryptsource $crypttarget --key-file=-"
+		if [ -n "$crypttpm" ]; then
+			cryptopen="$cryptopen --tpm=$crypttpm"
+		fi
+		if [ -n "$crypttpmpcr" ]; then
+			cryptopen="$cryptopen --tpmpcr=$crypttpmpcr"
+		fi
 	elif [ "$crypttcrypt" = "yes" ]; then
 		cryptopen="$cryptopen open --type tcrypt $cryptveracrypt $cryptsource $crypttarget"
 	else
